sn_cli/src/cli.rs

```diff
@@ -35,4 +35,8 @@ pub(crate) struct Opt {
     /// Available sub commands.
     #[clap(subcommand)]
     pub cmd: SubCmd,
+
+    /// Timeout in seconds for the CLI to wait for a data response from the network.
+    #[clap(long = "timeout", global = true)]
```

Comment:
I personally prefer keeping the parsing with clap. Like this:

```suggestion
    #[clap(long = "timeout", global = true, value_parser = |t: &str| -> Result<Duration> { Ok(t.parse().map(Duration::from_secs)?) })]
```

But then a `fn` instead of inlined.

Reply:
had no idea you could do that 👍